docs/DeveloperGuide.md

```diff
@@ -1,38 +1,146 @@
 # Developer Guide
 
+* [Acknowledgements](#acknowledgements)
+* [Setting up, getting started](#setting-up-getting-started)
+* [Design](#design)
+    * [Architecture](#architecture)
+    * [UI Component](#ui-component)
+    * [Parser Component](#parser-component)
+    * [Storage Component](#storage-component)
+    * [CardList Component](#cardlist-component)
+    * [TagList Component](#taglist-component)
+* [Implementation](#implementation)
+    * [card feature](#card-feature)
+    * [tag feature](#tag-feature)
+    * [export feature](#export-feature)
+    * [[Proposed] deck feature](#deck-feature)
+
+* [Appendix: Requirements](#appendix-requirements)
+    * [Product scope](#product-scope)
+    * [Target user profile](#target-user-profile)
+    * [Value proposition](#value-proposition)
+    * [User Stories](#user-stories)
+    * [Non-Functional Requirements](#non-functional-requirements)
+    * [Glossary](#glossary)
+
+---
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries --
+include links to the original source as well}
+
+---
+
+## Setting up, getting started
+
+---
+
+## Design
+
+* [Architecture](#architecture)
+* [UI Component](#ui-component)
+* [Parser Component](#parser-component)
+* [Storage Component](#storage-component)
+* [CardList Component](#cardlist-component)
+* [TagList Component](#taglist-component)
+
+### Architecture
+
+
+### UI Component
+API: Ui.java
+
+### Parser Component
+API: Parser.java
+
+### Storage Component
+API: Storage.java
+
+### CardList Component
+API: CardList.java
+
+### TagList Component
 
-## Design & implementation
+API: TagList.java
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## Implementation
 
+* [card feature](#card-feature)
+* [tag feature](#tag-feature)
+* [export feature](#export-feature)
+* [[Proposed] deck feature](#deck-feature)
+
+### Card Feature
+
+The current functionalities:
+* add card to the cardlist
+* delete card from the cardlist
+* show all cards the cardlist contains
+* add tag to a card
+* view the card by its uuid
+
+The implementation of the features are shown below: 
+
+* Adding a card
+1. When the user enters ```card add -q ... -a ...```, the input is passed to ```Parser``` class which calls ```Parser#parseCommand()```.
+2. The parser detects the keyword "card", then calls the ```Parser#CardKeywordParser()```.
+3. 
+
+The sequence diagram below shows how this feature works:
+{UML will be added here.}
```

Comment:
Would be great if there is some initial diagram (either class/sequence diagram) to help to illustrate the implementation.